### PR TITLE
Examples: Added BufferGeometry version of QRCode model

### PR DESCRIPTION
**Pull request by Mugen87**
see #14737

The model of the QR code is replaced with a `BufferGeometry` version. 

BTW: According to #14581, `SoftwareRenderer` does not support `BufferGeometry`. But the new model renders like the old one in `misc_ubiquity_test`. Is that a coincidence?

### CODE REVIEW

**Comment by Mugen87** [comment]
https://rawgit.com/Mugen87/three.js/dev17/examples/misc_ubiquity_test.html
https://rawgit.com/Mugen87/three.js/dev17/examples/svg_sandbox.html

File: examples/misc_ubiquity_test.html [hunk 1]
```diff
@@ -51,8 +51,8 @@
 
 				// QRCODE
 
-				var loader = new THREE.JSONLoader();
-				loader.load( 'models/json/QRCode.json', function ( geometry) {
+				var loader = new THREE.BufferGeometryLoader();
+				loader.load( 'models/json/QRCode_buffergeometry.json', function ( geometry ) {
 
 					mesh = new THREE.Mesh( geometry, new THREE.MeshLambertMaterial( { vertexColors: THREE.FaceColors } ) );
```
Comment (WestLangley) [inline examples/misc_ubiquity_test.html]: `FaceColors` do not make sense for `BufferGeometry`. You will have to track each renderer that uses `Projector.js` to see what hacks are implemented in that case.

Comment (Mugen87) [reply examples/misc_ubiquity_test.html]: Setting `VertexColors` does work now but the change feels like a hack. `Projector` and the respective renderers do not have a concept for supporting real vertex colors. Any chances to implement this different?

**Comment by WestLangley** [comment]
>Projector and the respective renderers do not have a concept for supporting real vertex colors for faces. Any chances to implement this different?

Unfortunately, `CanvasRenderer` does not support `THREE.VertexColors` for meshes; `BufferGeometry` does not support `THREE.FaceColors`.

We are going to have to figure out how we want to handle this....

Actually, I never liked the material property `vertexColors` with `THREE.FaceColors` as an option.

`vertexColors` should probably become a boolean.

**Comment by WestLangley** [comment]
Can you please modify `Projector` in a separate PR?

**Comment by Mugen87** [comment]
I've changed the PR so it only adds the new model.

**Comment by donmccurdy** [comment]
> Unfortunately, CanvasRenderer does not support THREE.VertexColors for meshes; BufferGeometry does not support THREE.FaceColors.

I hope I called it out clearly enough in the description that this wasn't missed and accidentally merged, but https://github.com/mrdoob/three.js/pull/14747 fixed that. CanvasRenderer will use the color of an arbitrary vertex in the face when `THREE.FaceColors` is set on a material applied to a BufferGeometry.  This lets us convert examples like `canvas / morphtargets / horse` to BufferGeometry.

**Comment by Mugen87** [comment]
You are referring to https://github.com/mrdoob/three.js/pull/14747/files#diff-12b5e517c94d02c07de80acaa6e6f320, right?

I think @WestLangley mentioned this point since it's a somewhat inconsistent to use `FaceColors` with a `BufferGeometry`. When this one is merged, I'll make a new PR with minor changes to `Projector` and the renderers so `VertexColors` can be used instead.

**Comment by donmccurdy** [comment]
Yes. I agree it's somewhat inconsistent, but `VertexColors` would also be inconsistent in a different way. Having one of the two work with BufferGeometry would be important, in any case.

**Comment by Mugen87** [comment]
@mrdoob I think this one can be merged. As mentioned in https://github.com/mrdoob/three.js/pull/14788#issuecomment-416296577, I plan a subsequent PR for this one.

**Comment by WestLangley** [comment]
@mrdoob There are multiple issues raised in this thread. What do you think?

**Comment by mrdoob** [comment]
Another option is to use multi-material.

**Comment by donmccurdy** [comment]
Yes, although that requires some rewriting of the rome horse model if we want to use it for the Canvas morph targets demo, and it will be a bit less efficient for WebGL rendering with more materials. To expand on the vertex / face colors question,

(A) (current behavior) CanvasRenderer looks for `THREE.FaceColors` on a material, and if found, chooses an arbitrary vertex color for each face, from that face's vertices.
  - Arguably somewhat reasonable from user perspective. Inconsistent with what WebGLRenderer would do with the same data, where face colors are not supported.

(B) CanvasRenderer looks for `THREE.VertexColors` on a material, and if found, chooses an arbitrary vertex color for each face, from that face's vertices.
  - Approximates what WebGLRenderer does with the same data, but visually inconsistent.

**Comment by Mugen87** [comment]
@donmccurdy `CanvasRenderer` could also look for both `THREE.FaceColors` and `THREE.VertexColors` and just do the same thing.

**Comment by Mugen87** [comment]
Although `CanvasRenderer` gone, it's still necessary to adjust `Projector` so it's possible to use the `BufferGeometry` version of `QRCode` with `SVGLoader`.

If this PR is merged, I'm going to apply the changes to `Projector`.

**Comment by WestLangley** [comment]
Do we need `QRCode`? What does including it demonstrate, exactly?

**Comment by Mugen87** [comment]
I think we could also use another model...

**Comment by WestLangley** [comment]
>Do we need QRCode? What does including it demonstrate, exactly?

If there is no answer to that question, I'd remove it from the examples.

**Comment by mrdoob** [comment]
Yeah, I think we can remove that example completely.

**Comment by WestLangley** [comment]
@mrdoob Remove which example(s)? Or do you mean remove the QRCode mesh?

**Comment by mrdoob** [comment]
> @mrdoob Remove which example(s)?

`misc_ubiquity_test.html`

Oh, but I can see it's also used in `svg_sandbox.html`. Okay, I guess we can keep the model and use @Mugen87's suggestion:

```js
if ( material.vertexColors === THREE.FaceColors ||  material.vertexColors === THREE.VertexColors ) {

    _face.color.fromArray( colors, a * 3 );

}
```

**Comment by Mugen87** [comment]
Okay, updated the PR. Ideally, it is merged together with #15114

**Pull request merged**